content/stub.completion-ui.js

```diff
@@ -84,15 +84,16 @@ ContactIdentityCompleter.prototype = {
     }
     // and since we can now map from contacts down to identities, map contacts
     //  to the first identity for them that we find...
-    matches = [val.NOUN_ID == Gloda.NOUN_IDENTITY ? val : val.identities[0]
-               for each ([iVal, val] in Iterator(contactToThing))];
+    matches = Array.prototype.map.call(contactToThing, function(val) {
+      return val.NOUN_ID == Gloda.NOUN_IDENTITY ? val : val.identities[0];
+    });
```

Comment:
While all the above fixes seem right, this one seems weird.

Here's a suggestion. For all the cases where Iterator is called with an Object that's abused as a key/value store, use entries.

* Copy/paste somewhere the entries function from https://esdiscuss.org/topic/es6-iteration-over-object-values
* Rewrite `for each ([k, v] in Iterator(obj))` into `for (let [k, v] of entries(obj))` and everything should be alright and easy!